qemu/qemu.go

```diff
@@ -704,6 +704,78 @@ func (blkdev BlockDevice) QemuParams(config *Config) []string {
 	return qemuParams
 }
 
+// VhostUserDeviceType is a qemu networking device type.
+type VhostUserDeviceType string
+
+const (
+	//VhostUserSCSI represents a SCSI vhostuser device type
+	VhostUserSCSI = "vhost-user-scsi-pci"
+	//VhostUserNet represents a net vhostuser device type
+	VhostUserNet = "virtio-net-pci"
+)
+
+// VhostUserDevice represents a qemu vhost-user network device meant to be passed
+// in to the guest
+type VhostUserDevice struct {
+	SocketPath    string //path to vhostuser socket on host
+	CharDevID     string
+	TypeDevID     string //id (SCSI) or netdev (net) device parameter
+	MacAddress    string //only valid if device type is  VhostUserNet
```

Comment:
Call it address. So that it can be reused for other types when applicable.

Reply:
Can we also add vhost-user-blk

```
-device vhost-user-blk-pci,logical_block_size=4096,size=512M,chardev=char0[,num_queues=N]
```

Having three may help us figure out what is generic and what is specific

Reply:
In the case of vhost-user-blk-pci, do you want to have 4k block and 512M size the default but configurable otherwise?